crates/fuel-core/src/schema/coins.rs

```diff
@@ -96,8 +96,10 @@ impl MessageCoin {
         self.0.amount.into()
     }
 
-    async fn asset_id(&self) -> AssetId {
-        fuel_core_types::fuel_types::AssetId::BASE.into()
+    async fn asset_id(&self, ctx: &Context<'_>) -> AssetId {
```

Comment:
Curious why this is called `asset_id` vs `base_asset_id`

Reply:
If I had to argue in favour of the name `asset_id` (`assetId` in GraphQL parlance), I would say that this naming gives a uniform interface on top of any coin type for retrieving the Asset ID. Specifically, we can query the `assetId` of a `Coin` or `MessgeCoin` without modifying our GQL queries. It also means we don't leak internal details: Calling it `base_asset_id` would be a leaky abstraction because it ties `MessageCoin` to the base layer, even though this is an irrelevant detail at the GQL layer. A user just wants the Asset ID of a given coin - they may not care if it's a regular coin or message coin.

On the other hand, calling it `base_asset_id` explicitly communicates its relation to the base layer. Maybe some users might like that. 

Personally, I'm in favour of keeping it `asset_id` for reasons above, but I am not attached one way or another. If we want to change it, I'm happy to do so.

Reply:
Makes sense given the context that this is on a MessageCoin, as it should be implicit that the amount of the asset is related to the base id.

However, from a user perspective it could also be confusing if they are trying to understand how to parse messages. Since if an erc20 is bridged, they may confuse the amount and asset with the erc20 coin they wanted to bridge, which would actually be encoded into the data section.

Since it's implicit anyways, do we even need the asset field here at all?

Reply:
Yeah, you could argue we don't even need it here. If it has a greater potential for added confusion, rather than for added benefit, we could axe it. If user's goal is to get the base asset ID, they can always get it from the consensus parameters:
<img width="548" alt="Screenshot 2023-09-13 at 5 42 27 PM" src="https://github.com/FuelLabs/fuel-core/assets/5528713/9755a6e7-3cba-4a45-ba93-c7a13b85a87f">

Removing it is a bit tangential to this PR, so I can do it here if we prefer, or in a follow up PR.